vcd/datasource_vcd_nsxt_network_context_profile_test.go

```diff
@@ -0,0 +1,134 @@
+	preTestChecks(t)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	skipNoNsxtConfiguration(t)
+
+	var params = StringMap{
+		"Org":             testConfig.VCD.Org,
+		"NsxtManagerName": testConfig.Nsxt.Manager,
+		"Scope":           "SYSTEM",
+		"Tags":            "nsxt network vdcGroup",
+	}
+
+	configText1 := templateFill(testAccVcdNsxtNetworkContextProfileNsxtManagerDS, params)
+	debugPrintf("#[DEBUG] CONFIGURATION - step 1: %s", configText1)
+
+	params["FuncName"] = t.Name() + "-step2"
+	params["Scope"] = "PROVIDER"
+	configText2 := templateFill(testAccVcdNsxtNetworkContextProfileNsxtManagerDS, params)
+	debugPrintf("#[DEBUG] CONFIGURATION - step 2: %s", configText2)
+
+	params["FuncName"] = t.Name() + "-step3"
+	params["Scope"] = "TENANT"
+	configText3 := templateFill(testAccVcdNsxtNetworkContextProfileNsxtManagerDS, params)
+	debugPrintf("#[DEBUG] CONFIGURATION - step 3: %s", configText3)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: configText1,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr("data.vcd_nsxt_network_context_profile.p", "id", regexp.MustCompile("urn:vcloud:networkContextProfile:")),
+					resource.TestCheckResourceAttr("data.vcd_nsxt_network_context_profile.p", "name", "CTRXICA"),
+					resource.TestCheckResourceAttr("data.vcd_nsxt_network_context_profile.p", "scope", "SYSTEM"),
+				),
+			},
+			{
+				Config: configText2,
+				// VCD has no capability to create items in PROVIDER context yet
+				ExpectError: regexp.MustCompile("entity not found"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr("data.vcd_nsxt_network_context_profile.p", "id", regexp.MustCompile("urn:vcloud:networkContextProfile:")),
+					resource.TestCheckResourceAttr("data.vcd_nsxt_network_context_profile.p", "name", "CTRXICA"),
+					resource.TestCheckResourceAttr("data.vcd_nsxt_network_context_profile.p", "scope", "PROVIDER"),
+				),
+			},
+			{
+				Config: configText3,
+				// VCD has no capability to create items in TENANT context yet
+				ExpectError: regexp.MustCompile("entity not found"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr("data.vcd_nsxt_network_context_profile.p", "id", regexp.MustCompile("urn:vcloud:networkContextProfile:")),
+					resource.TestCheckResourceAttr("data.vcd_nsxt_network_context_profile.p", "name", "CTRXICA"),
+					resource.TestCheckResourceAttr("data.vcd_nsxt_network_context_profile.p", "scope", "TENANT"),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdNsxtNetworkContextProfileNsxtManagerDS = `
+data "vcd_nsxt_manager" "main" {
+  name = "{{.NsxtManagerName}}"
+}
+
+data "vcd_nsxt_network_context_profile" "p" {
+  context_id = data.vcd_nsxt_manager.main.id
+  name       = "CTRXICA"
+  scope      = "{{.Scope}}"
+}
+`
```

Comment:
can context profile be from VDC Group?

Reply:
It can as per documentation, but there is no mechanism to create them in UI yet anywhere. It may be that API can do that, but not aiming for any "create" options until UI has it.

Excerpt from docs:
```
* Org vDC ID (_context==orgVdcId) - Returns all the network context profiles which are available to a specific Org vDC.
* Network provider ID (_context==networkProviderId) - | Returns all the network context profiles which are available for a specific network provider.
* VDC Group Id (_context==vdcGroupId) - | Returns all the network context profiles which are available to a specific vDC Group. 
```
https://developer.vmware.com/apis/vmware-cloud-director/latest/cloudapi/1.0.0/networkContextProfiles/get/

Reply:
I see, I was seeking test to have if we support VDC group :)

Reply:
Yep - to clarify. It is yet impossible to create any type of network context profiles. That is why this PR has only a data source.